aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -2517,17 +2540,13 @@ def register_route(func: AnyCallableT) -> AnyCallableT:
                 include_in_schema,
                 frozen_security,
                 fronzen_openapi_extensions,
+                deprecated,
             )
 
             # Collate Middleware for routes
             if middlewares is not None:
                 for handler in middlewares:
-                    if self._routes_with_middleware.get(route_key) is None:
-                        self._routes_with_middleware[route_key] = [handler]
-                    else:
-                        self._routes_with_middleware[route_key].append(handler)
-            else:
-                self._routes_with_middleware[route_key] = []
+                    self._routes_with_middleware[route_key].append(handler)
```

Comment:
I understand where you're going with this change (and the `defaultdict` definition) and that's really cool, but I don't know if we'll have any side effects if we don't define `self._routes_with_middleware[route_key] = []` and some check later will fail. 

I'm afraid that we are probably missing something in our tests.

Reply:
I understand, all good by me :+1: